pkg/machines/xmlCreator.js

```diff
@@ -85,13 +85,25 @@ export function getNetworkXML({ name, forwardMode, device, ipv4, netmask, ipv6,
         forwardMode === 'open') {
         const domainElem = doc.createElement('domain');
         domainElem.setAttribute('name', name);
+        domainElem.setAttribute('localOnly', 'yes');
         networkElem.appendChild(domainElem);
     }
 
     if (ipv4) {
+        const dnsElem = doc.createElement('dns');
+        const hostElem = doc.createElement('host');
+        hostElem.setAttribute('ip', ipv4);
+        const hostnameElem = doc.createElement('hostname');
+        const hostnameTextNode = doc.createTextNode('gateway');
```

Comment:
Can someone explain this here?
I usually use this xml syntax to assign static IPs to VMs with specific hostname, so where you put the gateway I usually put the VM's hostname?
I am sure this here is something different, @skobyda or @vasile-papp if you can please clarify this, otherwise it's good to get merged.

Reply:
Yes, the "dns" section is exactly for that purpose (associate "name" to "IP" or "IP" to "name"), this piece of code will just assign the "gateway" name to the "IP" address of the default gateway from the virtual network (in order to be resolved by the virtual dns).
```
  <dns>
    <host ip='192.168.100.1'>
      <hostname>gateway</hostname>
    </host>
  </dns>
```
You can continue to add new "hosts" in this section just as before.
```
  <dns>
    <host ip='192.168.100.1'>
      <hostname>gateway</hostname>
    </host>
    <host ip='192.168.100.100'>
      <hostname>vm1</hostname>
    </host>
    <host ip='192.168.100.123'>
      <hostname>vm2</hostname>
    </host>
  </dns>
```

Reply:
So basically in your machine you have a VM called gateway, and you want that IP assigned statically. What I don't understand is why should that we the default configuration for everyone using cockpit-machines?

Would you mind joining #cockpit on freenode so that we can finalize the discussion and if all clear merge this today?

Thanks in advance for your contribution and sorry it took ages to get this in.